contracts/Marketplace.sol

```diff
@@ -7,18 +7,23 @@ import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
 import "./Collateral.sol";
 import "./Proofs.sol";
 
+import "hardhat/console.sol";
+
 contract Marketplace is Collateral, Proofs {
   using EnumerableSet for EnumerableSet.Bytes32Set;
 
   type RequestId is bytes32;
   type SlotId is bytes32;
+  type ActiveSlotId is bytes32;
 
   uint256 public immutable collateral;
   MarketplaceFunds private funds;
   mapping(RequestId => Request) private requests;
   mapping(RequestId => RequestContext) private requestContexts;
   mapping(SlotId => Slot) private slots;
   mapping(address => EnumerableSet.Bytes32Set) private activeRequests;
+  mapping(RequestId => mapping(ActiveSlotId => EnumerableSet.Bytes32Set)) private activeSlots;
+  mapping(RequestId => uint8) private activeSlotsIdx;
```

Comment:
Perhaps you could encapsulate `activeSlots` and `activeSlotsId` into a separate contract or struct? Then you can test its logic separately.